### ISSUE TITLE
Getting Focusrite Scarlett solo gen 4 to work with Rocksmith 2014 Remastered

### ISSUE THREAD

**Issue by Oakydokyt** [mdias/rs_asio#457]
To start, a few months ago I purchased the Scarlett 2i2 gen 4 to use as a Rocksmith Real Tone cable replacement and I'm using rs_asio which works perfectly on computer 1 (HP EliteDesk 800 G4 SFF i5-8500 Win10). 

I recently purchased the Scarlett solo gen 4 for the same use but on another identical computer (computer 2) but the Focusrite USB ASIO driver errors and won't load the Focusrite USB ASIO driver according to the rs_asio_log. 

I did the same exact same install on computer 2 as I did on computer 1 and they are identical computers.

I can take the Scarlett solo from computer 2 and plug it into computer 1 and it works perfectly with no changes to files.

I copied all of the Rocksmith folder files (including the Rocksmith.exe and all subfolders) from computer 1 and pasted them into computer 2 and nothing changes, computer 2 still errors when loading rs_asio.

I did try using ASIO4ALL on computer 2 and it worked but the latency was too high where it rendered the program unusable.

Given my scenario do you know what would cause the Focusrite USB ASIO driver to not load on computer 2?  

Thank you and your help would be greatly appreciated.   

rs_asio-log from computer 1

0.000 [INFO]   - Wrapper DLL loaded (v0.7.1)
0.000 [INFO]  PatchOriginalCode
0.055 [INFO]  image crc32: 0xd1b38fcb
0.303 [INFO]  Patching CoCreateInstance
0.303 [INFO]  Patch_CallAbsoluteIndirectAddress - num locations: 2
0.304 [INFO]  Patching call at 00C02DBD
0.304 [INFO]  Patching call at 00E75278
0.304 [INFO]  Patching PortAudio MarshalStreamComPointers
0.304 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.304 [INFO]  Patching call at 00E7483F
0.304 [INFO]  Patching PortAudio UnmarshalStreamComPointers
0.304 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.304 [INFO]  Patching call at 00E748F4
0.304 [INFO]  Patching Two Guitar Tones Connected Message Box (num locations: 1)
0.304 [INFO]  Patching bytes at 007C0C5F
0.306 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
0.306 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
0.306 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
0.306 [INFO]  GetWineAsioInfo - Looking for "wineasio32.dll"...  Not found.
0.307 [INFO]  GetWineAsioInfo - Looking for "wineasio.dll"...  Not found.
0.308 [INFO]  AsioHelpers::FindDrivers
0.309 [INFO]    Ableton Push
0.309 [INFO]    Focusrite Thunderbolt ASIO
0.309 [INFO]    Focusrite USB ASIO
0.309 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: Focusrite USB ASIO
0.309 [INFO]  Creating AsioSharedHost - dll: c:\program files\focusrite\drivers\focusriteusbasio32.dll
0.315 [INFO]    ASIO input channels info:
0.315 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Input 1
0.315 [INFO]      1 - active: 0, channel: 1, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Input 2
0.315 [INFO]      2 - active: 0, channel: 2, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Loopback 1
0.315 [INFO]      3 - active: 0, channel: 3, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Loopback 2
0.315 [INFO]    ASIO output channels info:
0.315 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: Output 1
0.315 [INFO]      1 - active: 0, channel: 1, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: Output 2
0.315 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
0.315 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: Focusrite USB ASIO
0.315 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
0.315 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[1] requesting ASIO driver: Focusrite USB ASIO
0.315 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
0.315 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 1 render devices, 2 capture devices
0.315 [INFO]    hr: S_OK
0.315 [INFO]    *ppEndpoint: 10FB24A8
0.315 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
0.315 [INFO]    hr: S_OK
0.315 [INFO]    *ppEndpoint: 10FB2328
0.315 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
0.315 [INFO]    hr: S_OK
0.315 [INFO]    *ppDevices: 10D81408
0.315 [INFO]  {ASIO Out} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
0.315 [INFO]  {ASIO Out} DebugWrapperDevicePropertyStore::GetCount
0.315 [INFO]    *cProps: 2
0.315 [INFO]  {ASIO Out} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
0.315 [INFO]  {ASIO Out} DebugWrapperEndpoint::GetDataFlow
0.315 [INFO]    *pDataFlow: eRender
0.315 [INFO]  {ASIO Out} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 1
0.315 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::GetDevicePeriod
0.315 [INFO]  {ASIO Out} RSAsioAudioClient::~RSAsioAudioClient
0.315 [INFO]  {ASIO IN 0} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
0.315 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetCount
0.315 [INFO]    *cProps: 5
0.315 [INFO]  {ASIO IN 0} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
0.315 [INFO]  {ASIO IN 0} DebugWrapperEndpoint::GetDataFlow
0.315 [INFO]    *pDataFlow: eCapture
0.315 [INFO]  {ASIO IN 0} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 1
0.315 [INFO]  {ASIO IN 0} DebugWrapperAudioClient<struct IAudioClient3>::GetDevicePeriod
0.315 [INFO]  {ASIO IN 0} RSAsioAudioClient::~RSAsioAudioClient
0.316 [INFO]  {ASIO IN 1} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
0.316 [INFO]  {ASIO IN 1} DebugWrapperDevicePropertyStore::GetCount
0.316 [INFO]    *cProps: 5
0.316 [INFO]  {ASIO IN 1} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
0.316 [INFO]  {ASIO IN 1} DebugWrapperEndpoint::GetDataFlow
0.316 [INFO]    *pDataFlow: eCapture
0.316 [INFO]  {ASIO IN 1} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 1
0.316 [INFO]  {ASIO IN 1} DebugWrapperAudioClient<struct IAudioClient3>::GetDevicePeriod
0.316 [INFO]  {ASIO IN 1} RSAsioAudioClient::~RSAsioAudioClient
0.325 [INFO]  {ASIO Out} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 17
0.325 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Exclusive
0.325 [INFO]    rejecting IEEE Float as it's incompatible with current ASIO sample type ASIOSTInt32LSB
0.325 [INFO]    requested format is not supported
0.325 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.325 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Exclusive
0.325 [INFO]    rejecting IEEE Float as it's incompatible with current ASIO sample type ASIOSTInt32LSB
0.325 [INFO]    requested format is not supported
0.325 [INFO]    hr: AUDCLNT_E_UNSUPPORTED_FORMAT
0.325 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::IsFormatSupported - ShareMode: Exclusive
0.325 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::Initialize - ShareMode: Exclusive Flags: 40000 bufferDuration: 3ms periodicity: 3ms
0.325 [INFO]  {ASIO Out} RSAsioAudioClient::Initialize - host requested buffer duration: 3ms (144 frames)
0.325 [INFO]  {ASIO Out} WAVEFORMATEX
  wFormatTag: fffe
  nChannels: 2
  nSamplesPerSec: 48000
  nAvgBytesPerSec: 384000
  nBlockAlign: 8
  wBitsPerSample: 32
  cbSize: 22
  ext.SubFormat: KSDATAFORMAT_SUBTYPE_PCM
  ext.Samples: 24
  ext.dwChannelMask: 3
0.325 [INFO]  {ASIO Out} RSAsioAudioClient::Initialize - actual buffer duration: 1ms (48 frames)
0.325 [INFO]  AsioSharedHost::Setup - startCount: 0
0.325 [INFO]    post output ready: 0
0.325 [INFO]    ASIOBufferSize - min: 16 max: 1024 preferred: 48 granularity: 16
0.325 [INFO]    Creating ASIO buffers (2 out, 4 in)...
0.325 [INFO]  AsioSharedHost::AsioCalback_asioMessage - selector: 7 value: 0 | returning: 0
0.325 [INFO]  AsioSharedHost::AsioCalback_asioMessage - selector: 2 value: 0 | returning: 1
0.325 [INFO]  AsioSharedHost::AsioCalback_asioMessage - selector: 1 value: 3 | returning: 0
0.325 [INFO]  AsioSharedHost::AsioCalback_asioMessage - selector: 1 value: 5 | returning: 0
0.325 [INFO]  AsioSharedHost::AsioCalback_asioMessage - selector: 1 value: 4 | returning: 0
0.325 [INFO]  AsioSharedHost::AsioCalback_asioMessage - selector: 1 value: 6 | returning: 0
0.325 [INFO]  AsioSharedHost::AsioCalback_asioMessage - selector: 1 value: 15 | returning: 0
0.325 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::GetBufferSize
0.325 [INFO]    *pNumBufferFrames: 48
0.325 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::GetStreamLatency
0.325 [INFO]    latency: 3ms
0.325 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::GetService - riid: IID_IAudioRenderClient
0.325 [INFO]    returning render client
0.325 [INFO]  DebugDeviceEnum::RegisterEndpointNotificationCallback
0.327 [INFO]  Patched_PortAudio_MarshalStreamComPointers
0.327 [INFO]  Patched_PortAudio_UnmarshalStreamComPointers
0.327 [INFO]  {ASIO Out} DebugWrapperAudioClient3::QueryInterface riid: {F2D67F48-1977-4991-A3FC-A093835A7DC2}
0.327 [INFO]  {ASIO Out} DebugWrapperAudioClient2<struct IAudioClient3>::QueryInterface riid: {F2D67F48-1977-4991-A3FC-A093835A7DC2}
0.327 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::QueryInterface riid: {F2D67F48-1977-4991-A3FC-A093835A7DC2}
0.327 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::SetEventHandle
0.327 [INFO]  {ASIO Out} DebugWrapperRenderClient::GetBuffer NumFramesRequested: 48
0.327 [INFO]  {ASIO Out} DebugWrapperRenderClient::ReleaseBuffer NumFramesWritten: 48
0.327 [INFO]  {ASIO Out} RSAsioAudioClient::SwapBuffers
0.327 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::Start
0.327 [INFO]  AsioSharedHost::Start - enter startCount: 0
0.327 [INFO]    Starting ASIO stream...
0.690 [INFO]  AsioSharedHost::Start - leave startCount: 1
0.697 [INFO]  Focusrite USB ASIO - AsioSharedHost::AsioCalback_bufferSwitch - buffer switch 1
0.697 [INFO]  {ASIO Out} DebugWrapperRenderClient::GetBuffer NumFramesRequested: 48
0.697 [INFO]  {ASIO Out} DebugWrapperRenderClient::ReleaseBuffer NumFramesWritten: 48
0.697 [INFO]  {ASIO Out} RSAsioAudioClient::SwapBuffers
0.698 [INFO]  Focusrite USB ASIO - AsioSharedHost::AsioCalback_bufferSwitch - buffer switch 2
0.699 [INFO]  Focusrite USB ASIO - AsioSharedHost::AsioCalback_bufferSwitch - buffer switch 3 (not logging upcoming switches)
0.699 [INFO]  {ASIO Out} xrun
0.699 [INFO]  {ASIO Out} DebugWrapperRenderClient::GetBuffer NumFramesRequested: 48
0.699 [INFO]  {ASIO Out} DebugWrapperRenderClient::ReleaseBuffer NumFramesWritten: 48

rs_asio-log computer 2
0.000 [INFO]   - Wrapper DLL loaded (v0.7.1)
0.000 [INFO]  PatchOriginalCode
0.036 [INFO]  image crc32: 0xd1b38fcb
0.283 [INFO]  Patching CoCreateInstance
0.283 [INFO]  Patch_CallAbsoluteIndirectAddress - num locations: 2
0.283 [INFO]  Patching call at 00C02DBD
0.283 [INFO]  Patching call at 00E75278
0.284 [INFO]  Patching PortAudio MarshalStreamComPointers
0.284 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.284 [INFO]  Patching call at 00E7483F
0.284 [INFO]  Patching PortAudio UnmarshalStreamComPointers
0.284 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.284 [INFO]  Patching call at 00E748F4
0.284 [INFO]  Patching Two Guitar Tones Connected Message Box (num locations: 1)
0.284 [INFO]  Patching bytes at 007C0C5F
0.285 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
0.286 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
0.286 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
0.286 [INFO]  GetWineAsioInfo - Looking for "wineasio32.dll"...  Not found.
0.287 [INFO]  GetWineAsioInfo - Looking for "wineasio.dll"...  Not found.
0.288 [INFO]  AsioHelpers::FindDrivers
0.289 [INFO]    Ableton Push
0.289 [INFO]    Focusrite Thunderbolt ASIO
0.289 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: Ableton Push
0.289 [INFO]  Creating AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
0.297 [ERROR]  ASIO Error: No device is connected to the PC.
0.297 [INFO]  Destroying AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
0.297 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
0.297 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: Ableton Push
0.297 [INFO]  Creating AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
0.303 [ERROR]  ASIO Error: No device is connected to the PC.
0.303 [INFO]  Destroying AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
0.303 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
0.303 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[1] requesting ASIO driver: Ableton Push
0.303 [INFO]  Creating AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
0.309 [ERROR]  ASIO Error: No device is connected to the PC.
0.309 [INFO]  Destroying AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
0.309 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
0.309 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 0 render devices, 0 capture devices
0.309 [INFO]    hr: 80070490
0.309 [INFO]    *ppEndpoint: 00000000
0.309 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
0.309 [INFO]    hr: 80070490
0.309 [INFO]    *ppEndpoint: 00000000
0.309 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
0.309 [INFO]    hr: S_OK
0.309 [INFO]    *ppDevices: 0FC35D80
2.506 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
2.507 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
2.507 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
2.507 [INFO]  AsioHelpers::FindDrivers
2.507 [INFO]    Ableton Push
2.507 [INFO]    Focusrite Thunderbolt ASIO
2.507 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: Ableton Push
2.507 [INFO]  Creating AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
2.514 [ERROR]  ASIO Error: No device is connected to the PC.
2.514 [INFO]  Destroying AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
2.515 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
2.515 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: Ableton Push
2.515 [INFO]  Creating AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
2.521 [ERROR]  ASIO Error: No device is connected to the PC.
2.521 [INFO]  Destroying AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
2.521 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
2.521 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[1] requesting ASIO driver: Ableton Push
2.521 [INFO]  Creating AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
2.527 [ERROR]  ASIO Error: No device is connected to the PC.
2.527 [INFO]  Destroying AsioSharedHost - dll: c:\program files\ableton\push driver\x64\abletonpushasio.dll
2.528 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
2.528 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 0 render devices, 0 capture devices
2.528 [INFO]    hr: 80070490
2.528 [INFO]    *ppEndpoint: 00000000
2.528 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
2.528 [INFO]    hr: 80070490
2.528 [INFO]    *ppEndpoint: 00000000
2.528 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
2.528 [INFO]    hr: S_OK
2.528 [INFO]    *ppDevices: 0FB52C50
2.537 [INFO]  DebugDeviceEnum::RegisterEndpointNotificationCallback
34.297 [INFO]  DebugDeviceEnum::UnregisterEndpointNotificationCallback
34.464 [INFO]   - Wrapper DLL unloaded

**Comment by Oakydokyt** [comment]
Thanks for the feedback!

I checked the focusrite asio drivers on both computers and they are both the same driver version. It appears the Focusrite Control 2 and driver install is the same for the gen 4th 2i2 and Solo. I moved my Solo to computer 1 and did a new install on computer 2 for the 2i2. The Solo works perfect on computer 1 and the 2i2 on computer 2 has the same errors as the Solo on computer 2. This means the problem is probably not the rs_asio driver.

**Comment by Oakydokyt** [comment]
I made a change to the rs_asio.ini file and now have a new computer 2 rs_asio-log file:
0.000 [INFO]   - Wrapper DLL loaded (v0.7.1)
0.000 [INFO]  PatchOriginalCode
0.039 [INFO]  image crc32: 0xd1b38fcb
0.287 [INFO]  Patching CoCreateInstance
0.287 [INFO]  Patch_CallAbsoluteIndirectAddress - num locations: 2
0.287 [INFO]  Patching call at 00C02DBD
0.287 [INFO]  Patching call at 00E75278
0.287 [INFO]  Patching PortAudio MarshalStreamComPointers
0.287 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.287 [INFO]  Patching call at 00E7483F
0.287 [INFO]  Patching PortAudio UnmarshalStreamComPointers
0.287 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.287 [INFO]  Patching call at 00E748F4
0.287 [INFO]  Patching Two Guitar Tones Connected Message Box (num locations: 1)
0.287 [INFO]  Patching bytes at 007C0C5F
0.289 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
0.289 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
0.289 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
0.289 [INFO]  GetWineAsioInfo - Looking for "wineasio32.dll"...  Not found.
0.290 [INFO]  GetWineAsioInfo - Looking for "wineasio.dll"...  Not found.
0.291 [INFO]  AsioHelpers::FindDrivers
0.291 [INFO]    Ableton Push
0.291 [INFO]    Focusrite Thunderbolt ASIO
0.291 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: Focusrite USB ASIO
0.291 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
0.291 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: Focusrite USB ASIO
0.291 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
0.291 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[1] requesting ASIO driver: Focusrite USB ASIO
0.292 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
0.292 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 0 render devices, 0 capture devices
0.292 [INFO]    hr: 80070490
0.292 [INFO]    *ppEndpoint: 00000000
0.292 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
0.292 [INFO]    hr: 80070490
0.292 [INFO]    *ppEndpoint: 00000000
0.292 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
0.292 [INFO]    hr: S_OK
0.292 [INFO]    *ppDevices: 0F884B98
13.185 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
13.185 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
13.185 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
13.185 [INFO]  AsioHelpers::FindDrivers
13.186 [INFO]    Ableton Push
13.186 [INFO]    Focusrite Thunderbolt ASIO
13.186 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: Focusrite USB ASIO
13.186 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
13.186 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: Focusrite USB ASIO
13.186 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
13.186 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[1] requesting ASIO driver: Focusrite USB ASIO
13.186 [ERROR]  RSAsioDeviceEnum::UpdateAvailableDevices - failed.
13.186 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 0 render devices, 0 capture devices
13.186 [INFO]    hr: 80070490
13.186 [INFO]    *ppEndpoint: 00000000
13.186 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
13.186 [INFO]    hr: 80070490
13.186 [INFO]    *ppEndpoint: 00000000
13.186 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
13.186 [INFO]    hr: S_OK
13.186 [INFO]    *ppDevices: 0FC31F60
13.196 [INFO]  DebugDeviceEnum::RegisterEndpointNotificationCallback
47.087 [INFO]  DebugDeviceEnum::UnregisterEndpointNotificationCallback
47.257 [INFO]   - Wrapper DLL unloaded

Comparing the rs_asio-log on computer 1 (Rocksmith working with 4th gen Scarlett) I can see the first error starts here :
0.309 [INFO] Creating AsioSharedHost - dll: c:\program files\focusrite\drivers\focusriteusbasio32.dll

I looked on both computers and found the focusriteusbasio32.dll driver:
c:\program files\focusrite\drivers\focusriteusbasio32.dll 

I don't know why this focusriteusbasio32.dll file works on one computer but not the other.

**Comment by Oakydokyt** [comment]
I was able to get Rocksmith and rs_asio to work on computer 3 (yes, another exact copy of computer 1 & 2). All I did was a new install of steam, rocksmith, Focusrite 4th gen Focusrite Control 2 and rs_asio. The latency is great, I don't notice any, just like computer 1.
As for why computer 2 didn't work is still a mystery and I'm not going to further troubleshoot it. I will put that computer into service where it does not need Rocksmith. 
Thanks!
